src/qml/models/onboardingmodel.h

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2024 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+
+#ifndef BITCOIN_QML_MODELS_ONBOARDINGMODEL_H
+#define BITCOIN_QML_MODELS_ONBOARDINGMODEL_H
+
+
+#include <common/settings.h>
+#include <common/system.h>
+#include <chainparams.h>
+#include <clientversion.h>
+#include <interfaces/chain.h>
+#include <univalue.h>
+#include <validation.h>
+
+#include <string>
+
+#include <QObject>
+#include <QString>
+#include <QUrl>
+
+class OnboardingModel : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(QString getDefaultDataDirString READ getDefaultDataDirString CONSTANT)
```

Comment:
can a flag in qml options model be set without requiring node init?

Reply:
I think jarol is right. This seems like too much duplication with the Options model. The Node in the options model can just be allowed to be null and set later. That way it can gather up the values during onboarding and set them only when the node is setup after onboarding is complete.

Reply:
makes sense modified the code in a separate branch and will push into new PR once the UI one have been merged.